flow/layers/image_filter_layer.cc

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/image_filter_layer.h"
+
+namespace flutter {
+
+ImageFilterLayer::ImageFilterLayer(sk_sp<SkImageFilter> filter)
+    : filter_(std::move(filter)) {}
+
+void ImageFilterLayer::Preroll(PrerollContext* context,
+                               const SkMatrix& matrix) {
+  Layer::AutoPrerollSaveLayerState save =
+      Layer::AutoPrerollSaveLayerState::Create(context);
+  ContainerLayer::Preroll(context, matrix);
+
+  if (!context->has_platform_view && context->raster_cache &&
```

Comment:
I agree, there is a bunch of duplication in many of the layers. I'd say it would be a good side project when we get a chance, but I don't think you meant to try to refactor this as part of this PR?

Reply:
Sure, if that significantly slows down the progress of this PR, please feel free to do that in a future PR, and create a Github issue to track the progress so we won't forget it :smiley:

Reply:
Done: https://github.com/flutter/flutter/issues/47165